heartbeat/docs/reference/configuration/heartbeat-options.asciidoc

```diff
@@ -0,0 +1,448 @@
+The username for authenticating with the server. This setting is optional. 
+
+[[monitor-http-password]]
+===== password
+
+The password for authenticating with the server. This setting is optional.
+
+//Steffen: When are authentication credential required? If you are sending a request and expecting a response?
+
+[[monitor-http-tls-ssl]]
+===== ssl
+
+The TLS/SSL connection settings for use with the HTTPS endpoint. If you don't
+specify settings, the system defaults are used. 
+
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: http 
+  schedule: '@every 5s'
+  urls: ["http://myhost:80"]
+  ipv4: true
+  ipv6: true
+  mode: any
+  ssl:
+    certificate_authorities: ['/etc/ca.crt']
+    supported_protocols: ["TLSv1.0", "TLSv1.1", "TLSv1.2"]  
+-------------------------------------------------------------------------------
+
+
+[[monitor-http-check]]
+===== check
+
+An optional `request` to send to the remote host and the expected `response`. 
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: http 
+  schedule: '@every 5s'
+  urls: ["http://myhost:80"]
+  ipv4: true
+  ipv6: true
+  mode: any
+  check.request:
+    method: "GET"
+    headers:
+    body:
+  check.response:
+    status: 0
+    headers:
+    body:
+-------------------------------------------------------------------------------
+
+
+//Steffen: I need help filling out the above example with headers and body
+
+Under `check.request`, specify these options:
+
+*`method`*:: The HTTP method to use. Valid values are `"HEAD"`, `"GET"` and
+`"POST"`.
+*`headers`*:: A dictionary of additional HTTP headers to send.
+*`body`*:: Optional request body content.
+
+Under `check.response`, specify these options:
+
+*`status`*:: The expected status code. If this setting is not configured or
+it's set to 0, any status code other than 404 is accepted. 
+*`headers`*:: The required response headers. 
+*`body`*:: The required response body content.
+
+[[monitors-scheduler]]
+==== Scheduler Options
+
+You specify options under `scheduler` to control the behavior of the task
+scheduler.
+
+Example configuration: 
+
+[source,yaml]
+-------------------------------------------------------------------------------
+heartbeat.scheduler:
+  limit: 0
+  location: 'UTC-08:00'
+-------------------------------------------------------------------------------
+
+
+//Steffen: What is a reasonable limit to set in the example? Also, I just sort of guessed at what would be valid for location in my example.
+
+
+[[heartbeat-scheduler-limit]]
+===== limit
+
+The number of concurrent tasks that Heartbeat is allowed to execute. If set
+to 0, there is no limit. The default is 0. 
+
+[[heartbeat-scheduler-location]]
+===== location
+
+The timezone for the scheduler.
+
+//Steffen: Is the location a required setting? If not, what timezone does the scheduler default to?
```

Comment:
I'm inclined to not say much about this since we haven't tested the feature. We can add more detail after we've tested and know that we want people to use this.